lib/PhasarLLVM/DataFlowSolver/IfdsIde/Problems/IDEExtendedTaintAnalysis.cpp

```diff
@@ -0,0 +1,841 @@
+    /// iterators to be pointers
+
+    // NOLINTNEXTLINE(llvm-qualified-auto, readability-qualified-auto)
+    auto It = call->arg_begin();
+    // NOLINTNEXTLINE(llvm-qualified-auto, readability-qualified-auto)
+    auto End = call->arg_end();
+    // NOLINTNEXTLINE(llvm-qualified-auto, readability-qualified-auto)
+    auto FIt = DestFun->arg_begin();
+    // NOLINTNEXTLINE(llvm-qualified-auto, readability-qualified-auto)
+    auto FEnd = DestFun->arg_end();
+
+    const std::string CalleeName =
+        (DestFun->hasName()) ? DestFun->getName().str() : "none";
+    LOG_IF_ENABLE(BOOST_LOG_SEV(lg::get(), DEBUG)
+                  << "##Call-FF at: " << psr::llvmIRToString(call)
+                  << " to: " << CalleeName);
+    for (; FIt != FEnd && It != End; ++FIt, ++It) {
+      auto From = makeFlowFact(It->get());
+      /// Pointer-Arithetics in the last indirection are irrelevant for
+      /// equality comparison. Argumentation similar to StoreFF
+      if (equivalentExceptPointerArithmetics(From, Source)) {
+        LOG_IF_ENABLE(BOOST_LOG_SEV(lg::get(), DEBUG)
+                      << ">\tmatch: " << From << " vs " << Source);
+        Ret.insert(transferFlowFact(Source, From, &*FIt));
+      } else {
+        LOG_IF_ENABLE(BOOST_LOG_SEV(lg::get(), DEBUG)
+                      << ">\tno match: " << From << " vs " << Source);
+      }
+    }
+    ptrdiff_t Offs = 0;
+
+    if (!VA) {
+      return Ret;
+    }
+
+    for (; It != End; ++It) {
+      auto From = makeFlowFact(It->get());
+      if (equivalentExceptPointerArithmetics(From, Source)) {
+        auto To = transferFlowFact(Source, From, VA);
+
+        Ret.insert(FactFactory.withIndirectionOf(To, {Offs}));
+        /// Model varargs as an additional aggregate-parameter. The varargs
+        /// are stored in contiguous memory one after the other. Ignore
+        /// padding for now.
+      }
+      Offs +=
+          ptrdiff_t(DL.getTypeAllocSize(It->get()->getType()).getFixedSize());
+    }
+#ifdef XTAINT_DIAGNOSTICS
+    allTaintedValues.insert(ret.begin(), ret.end());
+#endif
+    return Ret;
+  });
+}
+
+IDEExtendedTaintAnalysis::FlowFunctionPtrType
+IDEExtendedTaintAnalysis::getRetFlowFunction(n_t CallSite, f_t CalleeFun,
+                                             n_t ExitStmt,
+                                             [[maybe_unused]] n_t RetSite) {
+  LOG_IF_ENABLE(BOOST_LOG_SEV(lg::get(), DEBUG)
+                << "##Return-FF at: " << psr::llvmIRToString(CallSite));
+  const auto *Call = llvm::cast<llvm::CallBase>(CallSite);
+  return makeLambdaFlow<d_t>([this, Call, CalleeFun,
+                              ExitStmt{llvm::cast<llvm::ReturnInst>(ExitStmt)}](
+                                 d_t Source) {
+    std::set<d_t> Ret;
+
+    /// Don't qualify the 'auto' here, because we should not rely on those
+    /// iterators to be pointers
+
+    // NOLINTNEXTLINE(llvm-qualified-auto, readability-qualified-auto)
+    auto It = Call->arg_begin();
```

Comment:
We could just define this as the type it is `User::op_iterator`, so it matches what CallBase defines as it's iterator.